truss/remote/baseten/core.py

```diff
@@ -271,7 +272,7 @@ def create_truss_service(
         return model_version_json["id"], model_version_json["version_id"]
 
     if model_id is None:
-        if environment:
+        if environment != PRODUCTION_ENVIRONMENT_NAME:
```

Comment:
nit: can you add unit tests for this method?

Reply:
done!